### DISCUSSION TITLE
Creating Sales Order based on variable BOM

### DISCUSSION THREAD

**Discussion by akaney83** [inventree/InvenTree#2693] [Q&A] [answered]
I just came across this project, and am still trying to figure it all out, so I apologize if I am missing something.

I'd like to be able to create a BOM. Let's use a table for example. The table uses 4 legs, screws, and a tabletop, but the tabletop is customizable. It is built from a 4' wide piece of wood, but the length is variable. The input units would be inches, and the length would be the qty. 

Can I create this BOM and have the price auto-calculated and display someplace on the part?

Also, it would be nice to be able to add a markup to this part, so that a sell amount could be auto-calculated when a part is added to the sales order.

**Comment by matmair** [comment, marked as answer]
Hi there @hammsbeerman

We currently do not have a feature to auto-price parts based on attributes.
Due to the complexity of pricing in manufacturing I doubt there will be a satisfactory solution any time soon.
My suggestion would be to generate static parts with help of the API: https://inventree.readthedocs.io/en/latest/extend/python/
Calculate the prices at that time and just add them.

We are always open to feature requests. However how fast something is implemented depends on the interest by the devs.
